core/bin/oauth.rs

```diff
@@ -203,6 +204,68 @@ async fn connections_access_token(
     }
 }
 
+#[derive(Deserialize)]
+struct CredentialPayload {
+    provider: CredentialProvider,
+    credentials: serde_json::Map<String, serde_json::Value>,
+}
+
+async fn oauth_service_create_credential(
```

Comment:
please don't depart from the naming scheme of this file :pray: 

Should be `credentials_create`

Reply:
Unintended, copilot wrote it and I overlook that it wasn't consistent.
Will fix.